waspc/src/Wasp/Generator/Common.hs

```diff
@@ -6,35 +6,30 @@ module Wasp.Generator.Common
   )
 where
 
-import qualified Wasp.SemanticVersion as SV
+import qualified Wasp.SemanticVersion as V
 
 -- | Directory where the whole web app project (client, server, ...) is generated.
 data ProjectRootDir
 
 -- | Range of node versions that node packages generated by this generator work correctly with.
-nodeVersionRange :: SV.Range
+nodeVersionRange :: V.Range
 nodeVersionRange =
-  SV.rangeFromVersionsIntersection
-    [ (SV.BackwardsCompatibleWith, latestLTSVersion),
-      (SV.LessThanOrEqual, latestLTSExactVersionThatWeKnowWorks)
-    ]
+  V.Range [V.caret latestLTSVersion <> V.lte latestLTSExactVersionThatWeKnowWorks]
   where
-    latestLTSVersion = SV.Version 16 0 0
+    latestLTSVersion = V.Version 16 0 0
     -- There is a bug in node 16.15.1 (more correctly, in npm 8.11.0 that comes with it)
-    -- that messes up how Wasp uses Prisma. That is why we limited ourselves to <=16.15.0 for now.
+    -- that messes up how Wasp uses Prisma. That is why we limited ourselves to
+    -- <=16.15.0 for now.
     -- Bug issue on NPM CLI repo: https://github.com/npm/cli/issues/5018 .
-    latestLTSExactVersionThatWeKnowWorks = SV.Version 16 15 0
+    latestLTSExactVersionThatWeKnowWorks = V.Version 16 15 0
 
 -- | Range of npm versions that Wasp and generated projects work correctly with.
-npmVersionRange :: SV.Range
+npmVersionRange :: V.Range
 npmVersionRange =
-  SV.rangeFromVersionsIntersection
-    [ (SV.BackwardsCompatibleWith, latestLTSVersion),
-      (SV.LessThanOrEqual, latestLTSExactVersionThatWeKnowWorks)
-    ]
+  V.Range [V.caret latestLTSVersion <> V.lte latestLTSExactVersionThatWeKnowWorks]
```

Comment:
Why are we using `<>` instead of `,`? If it's to ensure that the we reduce the set to a minimal union, then my other comment points out a bug. Also, I'd probably vote for a smart constructor to ensure there's no way to use it incorrectly.:
```haskell
npmVersionRange = V.makeRange [V.caret latestLTSVersion, V.lte latestLTSExactVersionThatWeKnowWorks]
```

Reply:
Those are two different operations -> `<>` acts as on two comparator sets and produces a new, single comparator set, that contains comparators of both sets, so effectively it is `&&`, while `Range [cs, cs]` acts as || between two comparator sets.

Check this test for an illustration:
```hs
      show
        ( Range [
            lte (Version 1 3 6) <> backwardsCompatibleWith (Version 1 2 0),
            eq (Version 1 2 3)
          ]
        )
        `shouldBe` "<=1.3.6 ^1.2.0 || =1.2.3"

```

Reply:
Got it!